File: tools/extract_pdf.py
import sys
from pathlib import Path
try:
    import PyPDF2
except Exception as e:
    print('MISSING_PYPDF2')
    raise

if len(sys.argv) < 2:
    print('Usage: extract_pdf.py <pdf-path>')
    sys.exit(2)

pdf_path = Path(sys.argv[1])
if not pdf_path.exists():
    print(f'ERROR: file not found: {pdf_path}')
    sys.exit(3)

out_lines = []
with open(pdf_path, 'rb') as f:
    reader = PyPDF2.PdfReader(f)
    for i, page in enumerate(reader.pages):
        text = page.extract_text()
        out_lines.append(f'--- PAGE {i+1} ---')
        out_lines.append(text or '')

print('\n'.join(out_lines))

File: tools/extract_pdf.js
const fs = require('fs');
let pdf = require('pdf-parse');
if (typeof pdf !== 'function' && pdf && typeof pdf.default === 'function') {
  pdf = pdf.default;
}

const pdfPath = process.argv[2];
if (!pdfPath) {
  console.error('Usage: node extract_pdf.js <pdf-path>');
  process.exit(2);
}
try {
  const dataBuffer = fs.readFileSync(pdfPath);
  pdf(dataBuffer).then(function(data) {
    console.log(data.text || '');
  }).catch(err => {
    console.error('ERROR_PDF_PARSE', err);
    process.exit(1);
  });
} catch (err) {
  console.error('ERROR_READ', err && err.message);
  process.exit(1);
}
